py3status/modules/velib_metropole.py

```diff
@@ -0,0 +1,314 @@
+        """
+        reduce the zone to reduce the size of fetched data on refresh
+        """
+
+        latitudes = []
+        longitudes = []
+
+        for x in data:
+            latitudes.append(float(data[x]["station_gps_latitude"]))
+            longitudes.append(float(data[x]["station_gps_longitude"]))
+
+        self.gps_dict.update(
+            {
+                "gpsTopLatitude": max(latitudes),
+                "gpsTopLongitude": max(longitudes),
+                "gpsBotLatitude": min(latitudes),
+                "gpsBotLongitude": min(longitudes),
+            }
+        )
+        self.optimal_area = True
+
+    def _get_velib_data(self):
+        try:
+            return self.py3.request(
+                VELIB_ENDPOINT, params=self.gps_dict, timeout=self.request_timeout
+            ).json()
+        except self.py3.RequestException:
+            return None
+
+    def _manipulate(self, data):
+        new_data = {}
+        stations = []
+
+        for code in self.station_codes:
+            new_station = {}
+
+            # search station
+            station = next(
+                (item for item in data if int(item["station"]["code"]) == int(code)),
+                None,
+            )
+
+            # flat, camel to snake and cast...
+            for key, value in self.py3.flatten_dict(station, delimiter="_").items():
+                key = self._camel_to_snake_case(key)
+                if key in self.placeholders:
+                    new_station[key] = self._cast_number(value)
+
+            # station_due_date_s: station due date in dateime iso format
+            if all("station_due_date" in list for list in [new_station, self.placeholders]):
+                station_due_date = datetime.fromtimestamp(
+                    new_station["station_due_date"]
+                )
+                new_station.update({"station_due_date_s": station_due_date.isoformat()})
+
+            # station_state_code: station code for thresholds
+            if all("station_state" in list for list in [new_station, self.placeholders]):
+                new_station.update(
+                    {
+                        "station_state_code": int(
+                            self.station_states[new_station["station_state"]]
+                        )
+                    }
+                )
+
+            stations.append(new_station)
+
+        # forge return
+        for index, station in enumerate(stations, 1):
+            new_data[index] = station
+
+        return new_data
+
+    def velib_metropole(self):
+        # refresh
+        current_time = time()
+        refresh = current_time >= self.idle_time
+
+        # time
+        if refresh:
+            self.idle_time = current_time + self.cache_timeout
+            cached_until = self.cache_timeout
+        else:
+            cached_until = self.idle_time - current_time
+
+        if not self.toggled and not refresh:
+            self.toggled = False
```

Comment:
Your toggle is a bit off. This should be `if self.toggled and not refresh`. Maybe just rename it to `self.scrolling` then fix `on_click` code so scrolling is true here instead of opposite.